### ISSUE TITLE
PlatformException(NoClassDefFoundError, java.lang.NoClassDefFoundError: software.solid.fluttervlcplayer.-$$Lambda$FlutterVlcPlayer$iNXqoCGYLpbNLrcEiwVj76XncDI, null, null)

### ISSUE THREAD

**Issue by mtrfnvwork** [solid-software/flutter_vlc_player#288]
When I try to run demo project I get the error:

```
I/art     ( 9504): Rejecting re-init on previously-failed class java.lang.Class<software.solid.fluttervlcplayer.-$$Lambda$FlutterVlcPlayer$iNXqoCGYLpbNLrcEiwVj76XncDI>
E/flutter ( 9504): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: PlatformException(NoClassDefFoundError, java.lang.NoClassDefFoundError: software.solid.fluttervlcplayer.-$$Lambda$FlutterVlcPlayer$iNXqoCGYLpbNLrcEiwVj76XncDI, null, null)
E/flutter ( 9504): #0      VlcPlayerApi.create (package:flutter_vlc_player_platform_interface/src/messages/messages.dart:610:7)
E/flutter ( 9504): <asynchronous suspension>
E/flutter ( 9504): #1      MethodChannelVlcPlayer.create (package:flutter_vlc_player_platform_interface/src/method_channel/method_channel_vlc_player.dart:55:12)
E/flutter ( 9504): <asynchronous suspension>
E/flutter ( 9504): #2      VlcPlayerController.initialize (package:flutter_vlc_player/src/vlc_player_controller.dart:182:5)
E/flutter ( 9504): <asynchronous suspension>
E/flutter ( 9504): #3      VlcPlayerController.onPlatformViewCreated (package:flutter_vlc_player/src/vlc_player_controller.dart:941:7)
E/flutter ( 9504): <asynchronous suspension>
E/flutter ( 9504): 
```

flutter doctor -v
```
[✓] Flutter (Channel stable, 2.8.1, on macOS 12.1 21C52 darwin-x64, locale
    ru-RU)
    • Flutter version 2.8.1 at ...
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 77d935af4d (13 days ago), 2021-12-16 08:37:33 -0800
    • Engine revision 890a5fca2e
    • Dart version 2.15.1

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at ...
    • Platform android-31, build-tools 32.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (3 available)
    • HTC U Play (mobile) • Android 6.0 (API 23)
    • iPhone 11 Pro Max (mobile)
    • Chrome (web)

• No issues found!
```

**Comment by nibha181** [comment]
I tried to run a video using flutter_vlc_player and getting the same issue. Did you get any solution on this?
```
E/flutter ( 6203):
E/flutter ( 6203): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: PlatformException(NoClassDefFoundError, java.lang.NoClassDefFoundError: software.solid.fluttervlcplayer.-$$Lambda$
FlutterVlcPlayer$iNXqoCGYLpbNLrcEiwVj76XncDI, null, null)
E/flutter ( 6203): #0      VlcPlayerApi.create (package:flutter_vlc_player_platform_interface/src/messages/messages.dart:610:7)
E/flutter ( 6203): <asynchronous suspension>
E/flutter ( 6203): #1      MethodChannelVlcPlayer.create (package:flutter_vlc_player_platform_interface/src/method_channel/method_channel_vlc_player.dart:55:12)
E/flutter ( 6203): <asynchronous suspension>
E/flutter ( 6203): #2      VlcPlayerController.initialize (package:flutter_vlc_player/src/vlc_player_controller.dart:182:5)
E/flutter ( 6203): <asynchronous suspension>
E/flutter ( 6203): #3      VlcPlayerController.onPlatformViewCreated (package:flutter_vlc_player/src/vlc_player_controller.dart:941:7)
E/flutter ( 6203): <asynchronous suspension>
```

**Comment by andrewcapatina** [comment]
+1, facing same issue. 

I can't run on physical device but emulator works. 

`W/ResourcesManager( 4784): getTopLevelResources: /data/app/com.example.ad_hoc.ad_hoc_25v-2/base.apk / 1.0 running in com.example.ad_hoc.ad_hoc_25v rsrc of package null
D/SecWifiDisplayUtil( 4784): Metadata value : none
I/art     ( 4784): Rejecting re-init on previously-failed class java.lang.Class<software.solid.fluttervlcplayer.-$$Lambda$FlutterVlcPlayer$iNXqoCGYLpbNLrcEiwVj76XncDI>
I/art     ( 4784): Rejecting re-init on previously-failed class java.lang.Class<software.solid.fluttervlcplayer.-$$Lambda$FlutterVlcPlayer$iNXqoCGYLpbNLrcEiwVj76XncDI>`

**Comment by kostov** [comment]
Same problem when running on Android 6.

**Comment by bailyzheng** [comment]
+1 Get the same problem when running on Android 6 while not on Android 10.

**Comment by kostov** [comment]
Working workaround (for me) is a using version 6.0.2 in pubspec.yaml:

flutter_vlc_player: 6.0.2 # WARNING: this version works on 6- Android

**Comment by kostov** [comment]
Found here in 6.0.5:

FlutterVlcPlayer.java, line 359
```
if (options != null) {
    options.forEach(option -> media.addOption(option)); // problem is here
}
```

Log: Rejecting re-init on previously-failed class java.lang.Class<software.solid.fluttervlcplayer.-$$Lambda$FlutterVlcPlayer$iNXqoCGYLpbNLrcEiwVj76XncDI>

then exception: PlatformException (PlatformException(NoClassDefFoundError, java.lang.NoClassDefFoundError: software.solid.fluttervlcplayer.-$$Lambda$FlutterVlcPlayer$iNXqoCGYLpbNLrcEiwVj76XncDI, null, null))

fix (for example):
```
if (options != null) {
    for (String option: options) // works everywhere
        media.addOption(option);
}
```

**Comment by kostov** [comment]
I think old Androids doesn't support List.forEach()-es.

**Comment by solid-software** [comment]
Should be fixed in 7.1.1
